libd/src/d/semantic/symbol.d

```diff
@@ -754,19 +754,79 @@ struct SymbolAnalyzer {
 			methodIndex = oldMethodIndex;
 		}
 
-		thisType = Type.get(i).getParamType(false, true);
-
+		thisType = Type.get(pass.object.getObject()).getParamType(false, true);
 		import std.conv;
 		auto name = i.name.toString(context);
 		manglePrefix = manglePrefix ~ to!string(name.length);
 
 		i.mangle = "I" ~ manglePrefix;
 
-		assert(d.members.length == 0, "Member support not implemented for interfaces yet");
-		assert(d.bases.length == 0, "Interface inheritance not implemented yet");
+		foreach(b; d.bases) {
+			import d.semantic.identifier : AliasResolver;
+			i.bases ~= AliasResolver!(function Interface (identified) {
+					static if(is(typeof(identified) : Symbol)) {
+						if(auto i = cast(Interface) identified) {
+							return i;
+						}
+					}
+					
+					static if(is(typeof(identified.location))) {
+						import d.exception;
+						throw new CompileException(identified.location, typeid(identified).toString() ~ " is not an interface.");
+					} else {
+						// for typeof(null)
+						assert(0);
+					}
+				})(pass).visit(b);
+		}
+
+
+		auto members = DeclarationVisitor(pass, AggregateType.Class).flatten(d.members, i);
+		Method[] methods;
+
+		foreach(m; members) {
+			if (auto method = cast (Method) m) { 
+				scheduler.require(method, Step.Signed);
+				// check for (non-static) method body
+				if(method.fbody) {
+					import d.exception;
+					throw new CompileException(method.location, "non-static or non-final method can't have a body in interface");
```

Comment:
For get about final here too. If you don't handle it in the code, do not mention in the message.